modules/ocf_backups/files/upload-to-box

```diff
@@ -0,0 +1,444 @@
+                            method='PUT')
+
+    return response['shared_link']['url']
+
+
+def make_box_folder(folder, auth):
+    """Make a folder on Box.com using the MKCOL HTTP request method"""
+    conn = pycurl.Curl()
+
+    conn.setopt(conn.CUSTOMREQUEST, 'MKCOL')
+    conn.setopt(conn.USERPWD, auth)
+    conn.setopt(conn.URL, BOX_DAV + folder)
+    conn.perform()
+
+    http_status = conn.getinfo(pycurl.HTTP_CODE)
+    conn.close()
+
+    if http_status == 405:
+        print('Could not create a folder, it already exists, continuing...')
+    elif http_status == 401:
+        print('401 error trying to create a folder, check your authentication!')
+        sys.exit(2)
+    elif http_status != 201:
+        print('Error creating folder, HTTP ' + str(http_status))
+        sys.exit(3)
+
+
+def print_progress(multi, stats, prev_total):
+    """Show an updating progress bar for all ongoing uploads"""
+    width = get_terminal_size((80, 20)).columns  # Default size as a fallback
+
+    # Move cursor to the top left using ANSII magic to overwrite previous lines
+    sys.stdout.write('\033[1;1H')
+
+    bytes_partial = 0
+    for conn in multi.handles:
+        if conn.upload:
+            print(conn.upload.get_progress(width))
+            bytes_partial += conn.upload.bytes_uploaded
+
+    bytes_total = stats['bytes_uploaded'] + bytes_partial
+
+    # Print statistics on upload time and current speed
+    time_stats = 'Time so far: {}  Speed: {:.0f} Mb/s'.format(
+        friendly_time(time.time() - stats['start_time']),
+        (bytes_total - prev_total) * 0.000008
+    )
+    print(time_stats + (width - len(time_stats)) * ' ')
+
+    # Print more stats, this time on amount uploaded
+    upload_stats = 'Uploaded {}/{} files and {}/{} ({:.2f}%) in total'.format(
+        stats['num_finished'],
+        stats['num_total'],
+        friendly_file_size(bytes_total),
+        friendly_file_size(stats['bytes_total']),
+        100 * bytes_total / stats['bytes_total']
+    )
+    print(upload_stats + (width - len(upload_stats)) * ' ')
+
+    # If there are less progress bars now, add blank rows to overwrite the
+    # bottom rows as necessary (to get rid of lines with outdated information)
+    blank_rows = len(multi.free)
+    for row in range(blank_rows):
+        print(' ' * width)
+
+    # Print the next update in a second, allow interrupts
+    thread = threading.Timer(1, print_progress, [multi, stats, bytes_total])
+    thread.daemon = True
+    thread.start()
+
+
+def reset_connection(multi, conn):
+    """Resets a pycurl connection object to be used again"""
+    conn.fp.close()
+    multi.remove_handle(conn)
+    conn.upload = None
+    conn.fp = None
+    conn.file_bytes = 0
+    multi.free.append(conn)
+
+
+def friendly_file_size(num_bytes):
+    """To convert bytes to human-readable units
+
+    Adapted from http://stackoverflow.com/a/1094933/1979001
+    """
+    for unit in ['', 'KB', 'MB', 'GB', 'TB', 'PB']:
```

Comment:
I really hope not as well, but it's there just in case someone tries to back up petabytes of something I suppose. Good luck to them, it's going to take like a year with our transfer rate. (~10 hours for 1.2 TB currently, so 1024 TB ~= 355 days)

Yeah, I really hope we never have petabytes of backups lol